src/devices/bus/ieee488/hp9122c.cpp

```diff
@@ -0,0 +1,363 @@
+// license:BSD-3-Clause
+// copyright-holders: Sven Schnelle
+/*********************************************************************
+
+    hp9122c.cpp
+
+    HP9122c dual floppy disk drive
+
+*********************************************************************/
+
+#include "emu.h"
+#include "hp9122c.h"
+#include "cpu/m6809/m6809.h"
+#include "machine/i8291a.h"
+#include "machine/wd_fdc.h"
+
+DEFINE_DEVICE_TYPE(HP9122C, hp9122c_device, "hp9122c", "HP9122C Dual High density disk drive")
+
+hp9122c_device::hp9122c_device(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock)
+	: device_t(mconfig, HP9122C, tag, owner, clock),
+	  device_ieee488_interface(mconfig, *this),
+	  m_cpu{*this , "cpu"},
+	  m_i8291a{*this, "i8291a"},
+	  m_fdc{*this, "mb8876"},
+	  m_floppy{*this, "floppy%u", 0},
+	  m_hpib_addr{*this , "HPIB address"},
+	  m_testmode{*this, "Testmode"},
+	  m_intsel{3},
+	  m_fdc_irq{false},
+	  m_i8291a_irq{false},
+	  m_fdc_drq{false},
+	  m_i8291a_drq{false},
+	  m_cpuirq{false},
+	  m_cpufirq{false},
+	  m_index_int{false},
+	  m_ds0{false},
+	  m_ds1{false}
+{
+}
+
+static INPUT_PORTS_START(hp9122c_port)
+	PORT_START("HPIB address")
+	PORT_CONFNAME(0x1f, 0x00 , "HPIB address")
+	PORT_CONFSETTING(0, "0")
+	PORT_CONFSETTING(1, "1")
+	PORT_CONFSETTING(2, "2")
+	PORT_CONFSETTING(3, "3")
+	PORT_CONFSETTING(4, "4")
+	PORT_CONFSETTING(5, "5")
+	PORT_CONFSETTING(6, "6")
+	PORT_CONFSETTING(7, "7")
+	PORT_CONFSETTING(7, "8")
+	PORT_CONFSETTING(7, "9")
+	PORT_CONFSETTING(7, "10")
+	PORT_CONFSETTING(7, "11")
+	PORT_CONFSETTING(7, "12")
+	PORT_CONFSETTING(7, "13")
+	PORT_CONFSETTING(7, "14")
+	PORT_CONFSETTING(7, "15")
+
+	PORT_START("Testmode")
+	PORT_CONFNAME(0x01, 0x00, "Testmode")
+	PORT_CONFSETTING(1, "Off")
+	PORT_CONFSETTING(0, "On")
+
+INPUT_PORTS_END
+
+ioport_constructor hp9122c_device::device_input_ports() const
+{
+	return INPUT_PORTS_NAME(hp9122c_port);
+}
+
+void hp9122c_device::device_start()
+{
+	for(auto floppy : m_floppy)
+		floppy->get_device()->setup_index_pulse_cb(floppy_image_device::index_pulse_cb(&hp9122c_device::index_pulse_cb, this));
+}
+
+void hp9122c_device::device_reset()
+{
+}
+
+void hp9122c_device::ieee488_eoi(int state)
+{
+	m_i8291a->eoi_w(state);
+}
+
+void hp9122c_device::ieee488_dav(int state)
+{
+	m_i8291a->dav_w(state);
+}
+
+void hp9122c_device::ieee488_nrfd(int state)
+{
+	m_i8291a->nrfd_w(state);
+}
+
+void hp9122c_device::ieee488_ndac(int state)
+{
+	m_i8291a->ndac_w(state);
+}
+
+void hp9122c_device::ieee488_ifc(int state)
+{
+	m_i8291a->ifc_w(state);
+}
+
+void hp9122c_device::ieee488_srq(int state)
+{
+	m_i8291a->srq_w(state);
+}
+
+void hp9122c_device::ieee488_atn(int state)
+{
+	m_i8291a->atn_w(state);
+}
+
+void hp9122c_device::ieee488_ren(int state)
+{
+	m_i8291a->ren_w(state);
+}
+
+WRITE_LINE_MEMBER(hp9122c_device::i8291a_eoi_w)
+{
+	m_bus->eoi_w(this , state);
```

Comment:
Please be consistent with spacing - in function calls elsewhere you have no space before comma, but in these member functions you're placing spaces before commas.